validators/validator_set.go

```diff
@@ -298,6 +298,11 @@ func (t *Topology) applyPromotion(performanceScore, rankingScore map[string]num.
 		}
 	}
 
+	// demoted tendermint validators are also ersatz so we need to add them
+	for _, id := range demotedFromTM {
```

Comment:
maybe I'm misremembering this, but won't they already be there by construction? IIRC promote would already return them with the correct status in remainingValidators and then the iteration in line 293 would already put them in ersatz. what am I missing?

Reply:
No, in `promote` its only when the number of slots reduces that we move validators from seriesA into seriesB (and `removedFromSeriesA` is left empty). When a true "demotion" happens the demoted validator is returned in `removedFromSeriesA` but not returned in `seriesB`.

Which feels intentional?

Reply:
it feels messy mostly, but that's on me